azurelinuxagent/common/osutil/bigip.py

```diff
@@ -287,7 +287,12 @@ def get_first_if(self):
         if retsize == (expected * struct_size):
             logger.warn(('SIOCGIFCONF returned more than {0} up '
                          'network interfaces.'), expected)
-        sock = buff.tostring()
+        try:
+            # Python 3.9 removed the tostring() method on arrays, tobytes() is the new alias
+            sock = buff.tostring()
+        except AttributeError:
```

Comment:
Instead of try-except, wouldn't it be better to have a check for python version instead? (To ensure we dont catch another error as part of this)

Reply:
I think AttributeError is already pretty specific:

> Raised when an attribute reference (see Attribute references) or assignment fails. 

From [here](https://docs.python.org/3/library/exceptions.html#AttributeError).
For example, we wouldn't catch TypeErrors so I don't think we are catching anything we're not supposed to here.

Reply:
But I think the main thing we're trying to catch here is just the issues raised by py3.9 right? I mainly dont want some other error to be masked by this check (not sure though the feasibility of another error being raised as an AttributeError tbh). It also makes the code more apparent on what we're trying to achieve rather than inferencing it behind some other check.

Reply:
+1: should be version specific, plus it should be somehow abstracted so that we do not have version checks peppered throughout the code

Reply:
I agree; should there be an alias in our `future.py`?

Reply:
Ok, adding a helper method to `future.py`